libs/tokens/src/pure/TokenLogo/index.tsx

```diff
@@ -83,25 +121,46 @@ export function TokenLogo({ logoURI, token, className, size = 36, sizeMobile }:
     return logoURI ? uriToHttp(logoURI) : []
   }, [logoURI, token])
 
-  const validUrls = useMemo(() => urls.filter((url) => !invalidUrls[url]), [urls, invalidUrls])
+  const validUrls = useMemo(() => urls && urls.filter((url) => !invalidUrls[url]), [urls, invalidUrls])
 
-  const currentUrl = validUrls[0]
+  const currentUrl = validUrls?.[0]
+
+  const onError = useCallback(() => {
+    if (!currentUrl) return
 
-  const onError = () => {
     setInvalidUrls((state) => ({ ...state, [currentUrl]: true }))
-  }
+  }, [currentUrl, setInvalidUrls])
 
   const initial = token?.symbol?.[0] || token?.name?.[0]
 
+  if (isLpToken) {
+    return (
+      <TokenLogoWrapper className={className} size={size} sizeMobile={sizeMobile}>
+        <LpTokenWrapper size={size}>
+          <div>
+            <TokenLogo noWrap token={tokensByAddress[token.tokens?.[0]]} size={size} sizeMobile={sizeMobile} />
+          </div>
+          <div>
+            <TokenLogo noWrap token={tokensByAddress[token.tokens?.[1]]} size={size} sizeMobile={sizeMobile} />
+          </div>
+        </LpTokenWrapper>
+      </TokenLogoWrapper>
+    )
+  }
```

Comment:
This is super cool!!!! so we use the same component that you tough it to represent pools!